spring-cloud-dataflow-module-deployers/spring-cloud-dataflow-module-deployer-cloudfoundry/src/main/java/org/springframework/cloud/dataflow/module/deployer/cloudfoundry/ModuleStatusBuilder.java

```diff
@@ -56,9 +56,15 @@ ModuleStatusBuilder withId(ModuleDeploymentId id) {
 		return this;
 	}
 
-	// TODO: Find out what the application state candidates are
 	private static ModuleStatus.State convertState(String state) {
 		switch (state) {
+			case "STARTED":
+			case "STARTING":
+			case "STOPPED":
```

Comment:
An app can be `STOPPED` because it has been created but not yet started (we can't create it directly as `STARTED`, so every app will be in this state for a while when it's first deployed), or because it is in the process of being destroyed (which is a short-lived state). So `undeployed` would be technically correct, as it describes either of those states, but `deploying` is probably more informative as `STOPPED` will generally only be encountered during app startup (i.e. module `deploying`).

I'd leave it as is, but if `undeployed` better fits your mental model then it's a reasonable change.